src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/OmniSharpVSDiagnostic.cs

```diff
@@ -15,5 +16,50 @@ internal class OmniSharpVSDiagnostic : Diagnostic
         // We need to override the "Tags" property because the basic Diagnostic Tags property has a custom JsonConverter that does not allow
         // VS extensions to tags.
         public new Container<OmniSharpVSDiagnosticTag>? Tags { get; set; }
+
+        /// <summary>
+        /// Gets or sets the project and context (e.g. Win32, MacOS, etc.) in which the diagnostic was generated.
+        /// </summary>
+        [JsonProperty("_ms_projects")]
+        public OmniSharpVSProjectAndContext[]? Projects { get; set; }
+
+        /// <summary>
+        /// Gets or sets an expanded description of the diagnostic.
+        /// </summary>
+        [JsonProperty("_ms_expandedMessage")]
+        public string? ExpandedMessage { get; set; }
+
+        /// <summary>
+        /// Gets or sets a message shown when the user hovers over an error. If null, then message
+        /// is used(use <see cref="OmniSharpVSDiagnosticTags.SuppressEditorToolTip"/> to prevent a tool tip from being shown).
```

Comment:
nit:
```suggestion
        /// is used (use <see cref="OmniSharpVSDiagnosticTags.SuppressEditorToolTip"/> to prevent a tool tip from being shown).
```